### PR TITLE
Fec 8696 contrib presets

### PR DESCRIPTION
**Pull request by esakal**

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -390,6 +390,12 @@ export default class Player extends FakeEventTarget {
    * @private
    */
   _resizeWatcher: ResizeWatcher;
+  /**
+   * Holds preset component factories
+   * @type {?PresetComponent}
+   * @private
+   */
+  _presetComponents: PresetComponent[];
```
Comment (yairans) [inline src/player.js]:
```suggestion
  _presetComponents: Array<PresetComponent>;
```
fix the the jsdoc accordingally

File: src/player.js [hunk 2]
```diff
@@ -916,6 +923,10 @@ export default class Player extends FakeEventTarget {
     return Utils.Object.mergeDeep({}, this._config);
   }
 
+  get presetComponents(): PresetComponent[] {
```
Comment (yairans) [inline src/player.js]:
```suggestion
  get presetComponents(): Array<PresetComponent> {
```

File: src/player.js [hunk 3]
```diff
@@ -916,6 +923,10 @@ export default class Player extends FakeEventTarget {
...
+    return [...this._presetComponents];
```
Comment (yairans) [inline src/player.js]:
```suggestion
    return this._presetComponents;
```

File: src/player.js [hunk 4]
```diff
@@ -1340,6 +1351,15 @@ export default class Player extends FakeEventTarget {
     setLogLevel(level, name);
   }
 
+  /**
+   * get new logger
+   * @param {*} name name
+   * @returns {void}
+   */
+  getLogger(name: string): void {
```
Comment (yairans) [inline src/player.js]: why this new api?

File: src/player.js [hunk 5]
```diff
@@ -1488,12 +1509,17 @@ export default class Player extends FakeEventTarget {
               if (typeof plugin.getMiddlewareImpl === 'function') {
                 plugin.name === 'bumper' ? middlewares.push(plugin.getMiddlewareImpl()) : middlewares.unshift(plugin.getMiddlewareImpl());
               }
+
+              if (typeof plugin.getPresetComponents === 'function') {
+                presetComponents.push(...(plugin.getPresetComponents() || []));
```
Comment (yairans) [inline src/player.js]:
```suggestion
                presetComponents.concat(plugin.getPresetComponents() || []);
```

File: flow-typed/types/preset.types.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+
+type PresetComponent = {
```
[the remark this answers is not in the document]
Comment (yairans) [reply flow-typed/types/preset.types.js]: I'm for Option 1. The hierarchy structure is easier to understand.

**Comment by kaltura-hooks** [comment]
Hi @esakal,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: flow-typed/types/preset.types.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (OrenMe) [reply flow-typed/types/preset.types.js]: what are the pros/cons of each, can you please explain?

[the remark this answers is not in the document]
Comment (esakal) [reply flow-typed/types/preset.types.js]: @yairnas can you please reply to Oren?

**Comment by esakal** [comment]
@OrenMe @yairans updated pr per your request

File: src/player.js (hunk 4 above)
Comment (esakal) [reply src/player.js]: removed

File: src/utils/index.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export * from './util';
+export {ResizeWatcher} from './resize-watcher';
```
Comment (OrenMe) [inline src/utils/index.js]: ResizeWatcher is an internal utility, and is exposed via event to app, we don't plan to expose it for actual usage a utility.

File: src/utils/resize-watcher.js [hunk 8]
```diff
@@ -19,6 +19,7 @@ class ResizeWatcher extends FakeEventTarget {
    * @returns {void}
    */
   destroy(): void {
+    // todo sakal Oren I think it should also remove all listeners, you might have memory leaks
```
Comment (OrenMe) [inline src/utils/resize-watcher.js]: disconnect does unobserv and clears watcher: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/disconnect
For iframe api we clear the `onresize` ourselves.
Which event listener is left?

File: src/assets/style.css [hunk 9]
```diff
@@ -61,7 +61,6 @@
   width: 100%;
   height: 100%;
   position: absolute;
-  visibility: hidden;
   z-index: -1;
```
Comment (OrenMe) [inline src/assets/style.css]: let's remove the border as well, and set bigger negative z-index for safety.
Good catch!

File: src/utils/index.js (hunk 7 above)
Comment (esakal) [reply src/utils/index.js]: I need this one to be used in playkit-js-ui to support side panels correctly, I will explain in our meeting.

File: src/utils/resize-watcher.js (hunk 8 above)
Comment (esakal) [reply src/utils/resize-watcher.js]: I might be wrong here, this class inherit from `FakeEventTarget` which handles its own queue of listeners. So while you unobserve the window you still hold reference to handler functions of components.

File: src/utils/index.js (hunk 7 above)
Comment (OrenMe) [reply src/utils/index.js]: OK, but I would advise you have a contingency plan as I don't see a reason to use more than one in the player core, and I see you already built upon it in UI manager.

File: src/utils/resize-watcher.js (hunk 8 above)
Comment (OrenMe) [reply src/utils/resize-watcher.js]: `FakeEventTarget` is the base object, it mimics the native DOM event target.
It is used by an event manager to register and unregister from listeners.
It is the responsibility of the listener invocation to unregister, which is done in player.js.

**Comment by OrenMe** [comment]
done in #387

**Pull request closed without merging**